Azure DDoS Protection/Script - View Protected Public IPs tool/Get-AzDDOSProtectedIPs.ps1

```diff
@@ -0,0 +1,248 @@
+    return $objectv
+}
+function Split-StringandReturn {
+    param(
+        [Parameter(Mandatory)]
+        [String]$stringinput,
+        [Parameter(Mandatory)]
+        [Int]$returnpart
+    )
+    $s = $stringinput.Split('/')
+    return $s[$returnpart]
+}
+function Get-AzVNetFromSubnetID {
+    param (
+        [Parameter(Mandatory)]
+        [String]$subnetid
+    )
+    $vnet = $subnetid.Split('/')
+    return $vnet[8]
+}
+function Get-AzDDOSProtectPlan {
+    param (
+        [Parameter(Mandatory)]
+        [String]$ddosplanID
+    )
+    $ddosplan = $ddosplanID.Split('/')
+    return $ddosplan[8]
+}
+function New-CSVReportFile {
+    param (
+        [Parameter(Mandatory)]
+        [String]$filepath
+    )
+    New-Item $filepath -type file -force
+    Set-Content $filepath 'PIP_Name,PIP_Address,PIP_Subscription,Resource_Group,Associated_Resource,Resource_Type,Associated_Resource_RG,VNet,DDOS_Enabled,DDOS_Plan'
+    Write-Host -ForegroundColor Green "Created $($filepathr)" 
+}
+function Clear-CreatedJSONFiles {
+    param (
+        [Parameter(Mandatory)]
+        [String]$filepathp,
+        [Parameter(Mandatory)]
+        [String]$filepathv
+    )
+    Write-Host -ForegroundColor Yellow "Removing created JSON files..." 
+    Remove-Item $filepathp -force
+    Remove-Item $filepathv -force
+    Write-Host -ForegroundColor Green "Removed JSON files $($filepathp) and $($filepathv)" 
+}
+#endregion functions
+#--------------------------------------------------------------------------------------------
+#region main
+# Check if the user is logged in
+Write-Host -ForegroundColor Yellow "Checking if there is an active Azure Context..." 
+$context = Get-AzContext
+if (!$context) {
+    Connect-AzAccount
+    $context = Get-AzContext
+}
+Write-Host -ForegroundColor Green "Context Retrieved Successfully." 
+Write-Host $context.Name
+# Get the PIP and VNet resources from all available Azure Subscriptions
+Get-PIPResources
+
+New-CSVReportFile -filepath $filepathr
+# Parse the PIP resrouces from the pipresources JSON file
+Write-Host -ForegroundColor Yellow "Parsing Public IP resources..." 
+Get-Content -Path $filepathp | ConvertFrom-Json | foreach {
+    $pipinfo += Get-IPConfigDetails -ipconfigtext $_.IpConfigurationText -pipName $_.Name -pipAddr $_.IpAddress -pipID $_.Id
+}
+# Parse the VNet resources from the vnetresources JSON file
+Write-Host -ForegroundColor Yellow "Parsing Virtual Network resources..." 
+Get-Content -Path $filepathv | ConvertFrom-Json | foreach {
+    if ($_.DdosProtectionPlan.Id -ne $null) { $dplan = Get-AzDDOSProtectPlan -ddosplanID $_.DdosProtectionPlan.Id } else { $dplan = "Not Enabled" }
+    $vnetinfo += Get-VnetDetails -vName $_.Name -vDDOSe $_.EnableDdosProtectionText -vDDOSp $dplan -uRG $_.ResourceGroupName
+}
+Write-Host -ForegroundColor Green "Finished parsing Public IP and Virtual Network resources" 
+# Loop through the PIP resources sorted by PIP subscription to build the report csv file
```

Comment:
if an existing report file is in the directory, it would give error (So only one can be run in a day). Either we give this prompt for user to remove existing report or add time stamp to the file to create a different name

Reply:
The export report filename is set to be the date that the report is run, if the file exists it will be replaced/overwritten. 

@tobystic I can change the file name to be date and time so that it is more unique and reduce possible conflicts or being overwritten